repo-agent/pkg/llm/claude_test.go

```diff
@@ -146,112 +141,90 @@ func TestClaudeRun(t *testing.T) {
 		envVarValue      string
 		expectedAPIKey   string
 		expectError      bool
 		expectedErrorMsg string
 	}{
 		{
-			name:      "In-cluster with valid secret",
-			inCluster: true,
-			secret: &corev1.Secret{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      AnthropicAPIKeySecretName,
-					Namespace: RepoAgentSystemNamespace,
-				},
-				Data: map[string][]byte{
-					AnthropicAPIKeySecretKey: []byte("test-secret-key"),
-				},
+			name: "API key from file",
+			setupFunc: func(tokensDir string) func() {
+				// Create a temporary file to simulate the mounted secret
+				filePath := filepath.Join(tokensDir, AnthropicAPIKeySecretKey)
+				if err := os.WriteFile(filePath, []byte("file-api-key"), 0600); err != nil {
+					t.Fatalf("Failed to write API key file: %v", err)
+				}
+				return func() { os.Remove(filePath) }
 			},
-			expectedAPIKey: "test-secret-key",
+			expectedAPIKey: "file-api-key",
 			expectError:    false,
 		},
 		{
-			name:             "In-cluster with missing secret",
-			inCluster:        true,
-			secret:           nil, // No secret
-			expectError:      true,
-			expectedErrorMsg: fmt.Sprintf("failed to get secret %s", AnthropicAPIKeySecretName),
+			name: "API key from environment variable (file not found)",
+			setupFunc: func(_ string) func() {
+				// We don't create the file, so it's not found
+				return func() {}
+			},
+			envVarSet:        true,
+			envVarValue:      "env-api-key",
+			expectedAPIKey:   "env-api-key",
+			expectError:      false,
+			expectedErrorMsg: "", // No error expected, as env var is fallback
 		},
 		{
-			name:      "In-cluster with secret missing key",
-			inCluster: true,
-			secret: &corev1.Secret{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      AnthropicAPIKeySecretName,
-					Namespace: RepoAgentSystemNamespace,
-				},
-				Data: map[string][]byte{
-					"other-key": []byte("some-value"),
-				},
+			name: "No API key (file not found and env var not set)",
+			setupFunc: func(_ string) func() {
+				// We don't create the file, so it's not found
+				return func() {}
 			},
+			envVarSet:        false,
+			expectedAPIKey:   "",
 			expectError:      true,
-			expectedErrorMsg: fmt.Sprintf("secret %s does not contain key '%s'", AnthropicAPIKeySecretName, AnthropicAPIKeySecretKey),
+			expectedErrorMsg: "API key not found in",
 		},
+
 		{
-			name:           "Outside cluster with env var set",
-			inCluster:      false,
-			envVarSet:      true,
-			envVarValue:    "test-env-key",
-			expectedAPIKey: "test-env-key",
+			name: "API key from file with leading/trailing whitespace",
+			setupFunc: func(tokensDir string) func() {
+				filePath := filepath.Join(tokensDir, AnthropicAPIKeySecretKey)
+				if err := os.WriteFile(filePath, []byte("  file-api-key  "), 0600); err != nil {
+					t.Fatalf("Failed to write API key file: %v", err)
+				}
+				return func() { os.Remove(filePath) }
+			},
+			expectedAPIKey: "file-api-key",
 			expectError:    false,
 		},
```

Comment:
It's good that you've updated the tests to reflect the new secret loading mechanism. The use of a temporary directory for the tokens is a clean way to test this.